module/helper/tilemap_renderer/shaders/geom2d.wgsl

```diff
@@ -0,0 +1,37 @@
+struct Color
+{
+  r : f32,
+  g : f32,
+  b : f32
+};
+
+struct PushConstant
+{
+  aspect_scale : vec2< f32 >,
+  translation : vec2< f32 >,
+  rotation_sin_cos : vec2< f32 >,
+  scale : vec2< f32 >,
+  color : Color,
+};
```

Comment:
Regarding your question about how to upload a uniform value for each object
In webgpu you have to create new render pipeline for each like in this example
https://webgpu.github.io/webgpu-samples/?sample=stencilMask

Just a downside of not having a global context that does everything for you